models/ed/R/met2model.ED2.R

```diff
@@ -198,58 +212,37 @@ met2model.ED2 <- function(in.path, in.prefix, outfolder, start_date, end_date, l
       CO2 <- c(rep(CO2[1], toff), CO2)[slen]
     }
 
-    ## build time variables (year, month, day of year)
-    skip <- FALSE
-    nyr <- floor(length(sec) * dt / 86400 / 365)
-    yr   <- NULL
-    doy  <- NULL
-    hr   <- NULL
-    asec <- sec
-    for (y in seq(year, year + nyr - 1)) {
-      diy <- PEcAn.utils::days_in_year(y, leap_year)
-      ytmp <- rep(y, udunits2::ud.convert(diy / dt, "days", "seconds"))
-      dtmp <- rep(seq_len(diy), each = day_secs / dt)
-      if (is.null(yr)) {
-        yr  <- ytmp
-        doy <- dtmp
-        hr  <- rep(NA, length(dtmp))
-      } else {
-        yr  <- c(yr, ytmp)
-        doy <- c(doy, dtmp)
-        hr  <- c(hr, rep(NA, length(dtmp)))
-      }
-      rng <- length(doy) - length(ytmp):1 + 1
-      if (!all(rng >= 0)) {
-        skip <- TRUE
-        PEcAn.logger::logger.warn(year, " is not a complete year and will not be included")
-        break
-      }
-      asec[rng] <- asec[rng] - asec[rng[1]]
-      hr[rng]   <- (asec[rng] - (dtmp - 1) * day_secs) / day_secs * 24
-    }
-    mo <- day2mo(yr, doy, leap_year)
-    if (length(yr) < length(sec)) {
-      rng <- (length(yr) + 1):length(sec)
-      if (!all(rng >= 0)) {
-        skip <- TRUE
-        PEcAn.logger::logger.warn(paste(year, "is not a complete year and will not be included"))
-        break
-      }
-      yr[rng]  <- rep(y + 1, length(rng))
-      doy[rng] <- rep(1:366, each = day_secs / dt)[1:length(rng)]
-      hr[rng]  <- rep(seq(0, length = day_secs / dt, by = dt / day_secs * 24), 366)[1:length(rng)]
-    }
-    if (skip) {
-      print("Skipping to next year")
-      next
-    }
```

Comment:
This is the code I did not understand at all. As far as I could tell, its main job was to determine the day of year (`doy`) and local time (`hr`) for the purposes of calculating the solar angle. I opted for a more straightforward approach to both of these variables in the new version, which effectively boils down to:

```r
doy <- floor(tdays) + 1
# %% means "the remainder of"
hr <- (tdays %% 1) * 24
```